config/abilities/provider_any.rb

```diff
@@ -40,6 +40,14 @@
       can [:show], ServiceRelatedEvent do |event|
         user.has_access_to_service?(event.try(:service) || event.try(:service_id))
       end
+
+      can :suspend, Account do | buyer |
+        buyer.can_suspend? && buyer.provider_account == account
+      end
+
+      can :resume, Account do | buyer |
+        buyer.can_resume? && buyer.provider_account == account
+      end
```

Comment:
Not really needed since having access to the screen means all this conditions are met. But I think it's descriptive, and master users have these abilities, so having them also for provider users makes it consistent.